### ISSUE TITLE
Great circle calculation error

### ISSUE THREAD

**Issue by joabakk** [SignalK/freeboard-sk#69]
The current measure feature measures rhumb line distance in km. For longer distances, this deviates quite a bit from great circle. As this feature is not so often in use, it could make sense to default to great circle (https://en.wikipedia.org/wiki/Haversine_formula). Also, NM is more commonly used than km

**Comment by rob42** [comment]
When I wrote the original I considered using great circle. I did some comparisons and its only significant at distances greater than 200 NM. Even then its a few decimals different. When i crossed the Tasman Sea (Hobart-Nelson - 1100+NM) the difference was about 10 miles (from memory).

**Comment by panaaj** [comment]
Desired Distance Units can be configured in `Settings` and measure will use these.
![image](https://user-images.githubusercontent.com/38519157/51783456-ec49b380-2189-11e9-9a25-fc046f7e46c7.png)

**Comment by panaaj** [comment]
Just an FYI the formula currently used is:
```
dlat= lat2(radians) - lat1(radians);
dlon= lon2(radians) - lon1(radians);
a= Math.pow( sin(dlat/2),2) + cos(lat1) * cos(lat2) * Math.pow(Math.sin(dlon/2) ,2)
c= 2 * arctan2(Math.sqrt(a), sqrt(1-a) ) // great circle distance in radians
m= (c * 6373) * 1000; // great circle distance in m
```

**Comment by joabakk** [comment]
Ok then this is a non-issue. I must have done something wrong in my napkin calcs...

**Comment by joabakk** [comment]
Hm hang on a bit, what is the source of that equation? I get panama-london 4760nm from sea-diatances.org but 5700nm  from freeboard... There are some fomulas in https://en.wikipedia.org/wiki/Great-circle_distance

**Comment by panaaj** [comment]
The formula currently in use is the Haversine formula which seems to be, from my reading at least, the one to use (is accurate to around 0.3%).

To make sure the implementation is correct I have compared the results with other on-line GC calculators and the answer is the same for a given set of points. 

Below is the function definition and an example used to compare with online calculators.

```
        console.log( this.distanceTo( 
            [138.6327,-34.8785], [147.2899,-42.9337] 
        ) );

       // *** result: distance in metres ****
       1166306.4220188455

    distanceTo(srcpt, destpt) {  
        const Rk= 6371; // mean radius of the earth (km)
        let lat1= Convert.degreesToRadians(srcpt[1]);
        let lon1= Convert.degreesToRadians(srcpt[0]);
        let lat2= Convert.degreesToRadians(destpt[1]);
        let lon2= Convert.degreesToRadians(destpt[0]);
        let dlat= lat2 - lat1;
        let dlon= lon2 - lon1;
        let a= Math.pow(Math.sin(dlat/2),2) + Math.cos(lat1) * Math.cos(lat2) * Math.pow(Math.sin(dlon/2),2);
        let z= 2 * Math.atan2(Math.sqrt(a),Math.sqrt(1-a)); 
        return (z * Rk) * 1000; // great circle distance in m
    }  
```

**Comment by joabakk** [comment]
OK then I rest my case. Thanks for looking into it. And you are right, this is actually Haversine GC.

**Comment by chrisWyth** [comment]
Something still isn't right, I get about 20% too much distance as measured from 24.367E 35.197N to 24.123 34.861. Should be 24nM, shows 28.7nM. Funny heading too.
![freeboard-511](https://user-images.githubusercontent.com/47893383/53251408-3d838f00-36c5-11e9-8be8-0325b3c031ba.png)

**Comment by panaaj** [comment]
Closed by [e4a4293](https://github.com/SignalK/freeboard-sk/commit/e4a4293642b2407bc393d9188e2fa34db085e09a)
